src/seer/automation/models.py

```diff
@@ -352,43 +365,81 @@ def from_event(cls, error_event: SentryEventData):
                         continue
                     crumb_details = BreadcrumbsDetails.model_validate(breadcrumb)
                     breadcrumbs.append(crumb_details)
+            elif entry.get("type") == "stacktrace":
+                stacktraces.append(Stacktrace.model_validate(entry.get("data", {})))
+            elif entry.get("type") == "message":
+                message = entry.get("data", {}).get("formatted", None)
+            elif entry.get("type") == "request":
+                request = RequestDetails.model_validate(entry.get("data", {}))
 
         return cls(
             title=error_event.get("title"),
             transaction_name=transaction_name,
             exceptions=exceptions,
             threads=threads,
             breadcrumbs=breadcrumbs,
+            message=message,
+            stacktraces=stacktraces,
+            request=request,
         )
 
     def format_event(self):
+        exceptions = self.format_exceptions()
+        breadcrumbs = self.format_breadcrumbs()
+        message = self.message if self.message and self.message not in self.title else ""
+        stacktraces = self.format_stacktraces()
+        request = self.format_request()
+
         return textwrap.dedent(
             """\
             {title} {transaction}
-            <exceptions>
+            {message}
             {exceptions}
-            </exceptions>
-            <breadcrumb_logs>
+            {stacktraces}
             {breadcrumbs}
-            </breadcrumb_logs>
+            {request}
             """
         ).format(
             title=self.title,
             transaction=f"(occurred in: {self.transaction_name})" if self.transaction_name else "",
-            exceptions=self.format_exceptions(),
-            breadcrumbs=self.format_breadcrumbs(),
+            message=f"\n<message>\n{message}\n</message>" if message.strip() else "",
+            exceptions=f"<exceptions>\n{exceptions}\n</exceptions>" if exceptions.strip() else "",
+            stacktraces=(
+                f"\n<stacktraces>\n{stacktraces}\n</stacktraces>" if stacktraces.strip() else ""
+            ),
+            breadcrumbs=(
+                f"\n<breadcrumb_logs>\n{breadcrumbs}\n</breadcrumb_logs>"
+                if breadcrumbs.strip()
+                else ""
+            ),
+            request=f"\n<http_request>\n{request}\n</http_request>" if request.strip() else "",
         )
```

Comment:
```suggestion
        ).strip()
```
Probably good to strip here, I see in your last langfuse traces a lot of whitespace

![CleanShot 2025-04-02 at 13 14 21@2x](https://github.com/user-attachments/assets/39c1f3c9-14c5-4f8f-90f2-bfc035385eea)